pkg/cmd/run/view/view.go

```diff
@@ -29,35 +29,54 @@ import (
 )
 
 type runLogCache interface {
-	Exists(string) bool
-	Create(string, io.Reader) error
-	Open(string) (*zip.ReadCloser, error)
+	Exists(key string) (bool, error)
+	Create(key string, r io.Reader) error
+	Open(key string) (*zip.ReadCloser, error)
 }
 
-type rlc struct{}
+type rlc struct {
+	cacheDir string
+}
+
+func (c rlc) Exists(key string) (bool, error) {
+	_, err := os.Stat(c.filepath(key))
+	if err == nil {
+		return true, nil
+	}
 
-func (rlc) Exists(path string) bool {
-	if _, err := os.Stat(path); err != nil {
-		return false
+	if errors.Is(err, os.ErrNotExist) {
+		return false, nil
 	}
-	return true
+
+	return false, fmt.Errorf("checking cache entry: %v", err)
 }
-func (rlc) Create(path string, content io.Reader) error {
-	err := os.MkdirAll(filepath.Dir(path), 0755)
+
+func (c rlc) Create(key string, content io.Reader) error {
+	out, err := os.Create(c.filepath(key))
 	if err != nil {
-		return fmt.Errorf("could not create cache: %w", err)
+		return fmt.Errorf("creating cache entry: %v", err)
 	}
+	defer out.Close()
```

Comment:
**Is `os.MkdirAll` not needed anymore?**

Since the cache directory path can change, it seems like we'd still want to make sure it exists before creating the run log cache, right?

Reply:
Yeh that's a fair point. Initially I thought that something would always create this earlier, but that in fact may not be true. I'll make sure to put this back.

Reply:
Added it back, thanks!

Reply:
So, this is a whole topic as https://groups.google.com/g/golang-dev/c/TXlDtwxaTcM?pli=1 highlights as different operating systems behave different like umask ... 🤔